src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs

```diff
@@ -143,7 +144,7 @@ public bool NormalizeAndValidateFraction()
                 // .099999999   normalize to 1,000,000 ticks
 
                 Debug.Assert(_zeroes > 0); // Already validated that in the condition _zeroes == 0 && _num > MaxFraction
-                _num = (int)Math.Round((double)_num / Pow10(totalDigitsCount - MaxFractionDigits), MidpointRounding.AwayFromZero);
+                _num = (int)Math.Round(_num / Math.Pow(10, totalDigitsCount - MaxFractionDigits), MidpointRounding.AwayFromZero);
```

Comment:
```suggestion
                uint power = (uint)Pow10(totalDigitsCount - MaxFractionDigits);
                _num = (int)(((uint)_num + power / 2) / power);
```

Reply:
We have to be careful to ensure that `totalDigitsCount - MaxFractionDigits` is less than or equal `MaxFractionDigits` , otherwise we cannot call `Pow10`.

Reply:
This holds as long as `_zeros <= MaxFractionDigits`.

Reply:
I changed according to @xtqqczze suggeestion, and added a branch for case when `_zeros > MaxFractionDigits`. Also documented with an assert that `totalDigitsCount - MaxFractionDigits <= MaxFractionDigits` in other cases.